src/inbox/legacyReport.js

```diff
@@ -0,0 +1,40 @@
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { RestLink } from 'apollo-link-rest';
+import gql from 'graphql-tag';
+import { ApolloClient } from 'apollo-client';
+import usePromise from 'react-use-promise';
+import { useMemo } from 'react';
+
+// setup your `RestLink` with your endpoint
+const hackeroneRestLink = new RestLink({
+  uri: 'https://ngftg30rl3.execute-api.eu-central-1.amazonaws.com/prod/',
+});
+
+// setup your client
+const hackeroneRestClient = new ApolloClient({
+  link: hackeroneRestLink,
+  cache: new InMemoryCache(),
+});
+
+const restReportQuery = gql`
+  query restReport($path: String!) {
+    report @rest(type: "RestReport", path: $path) {
+      databaseId: id
+      substate
+      title
+    }
+  }
+`;
+
+const useEnhancedReport = reportId => usePromise(
+  useMemo(
+    () => {
+      console.log(`MEMO => ${reportId}`);
```

Comment:
For some reason, this method returns the wrong report data and I'm not sure why yet. Initially, I thought it was due to the missing variables on memo's second argument but that wasn't the case. I think it has to do with the rest technique and the cache we're using, it might not use the right cache key and therefore reusing the wrong item.